doc/src/manual/functions.md

```diff
@@ -1159,6 +1159,9 @@ julia> 1:5 .|> [x->x^2, inv, x->2*x, -, isodd]
  true
 ```
 
+!!! warning
+    It is always important to understand the mechanisms behind the magic of unique language features like fused broadcast loops. There are some occasional pitfalls which are important to avoid. For an example on when *not* to fuse broadcasted operations, see the note in [Performance Tips](@ref man-performance-unfuse). Users should also remember that due to the in-place mutation behavior of the `.=` operator, extra care should be taken to ensure correct and intended semantics when an object appears in both arguments of the operator like `x .= foo.(x)` or similar expressions.
```

Comment:
I think the basic point here is about how many times the function are called. I don't think it should immediately be highlighted as a problem, and I think we can explain what goes right here, so that you get the right mental model. 

> When some of the arguments are smaller than the final broadcast result, all functions in the fused broadcast are still called for every element of the result. Thus `x .+ σ .* randn.()` adds a different random number to every element of the array `x`. But also, something like `y .= x ./ sqrt.(s)` will call `sqrt` all `length(y)` times, even though it has only `length(s)` distinct values. It may be faster to allocate space & save computation by avoiding fusion in some such cases -- see more in [performance tips](...).

The bit about in-place broadcasting seems to be hinting at aliasing, but IMO that's a different paragraph, and it needs to tell you facts.

Reply:
I am certainly aiming for a neutral explanation and don't mean to highlight problems that aren't there. I do want to check --- is your comment directed at the wording in the `!!! warning` under `functions.md` or directed at the blurb in Performance Tips? If the latter, I would protest slightly, as the wording (emph. added)

> # Fewer dots: Unfuse _certain_ intermediate broadcasts

> However, it is important to remember that the fused operation will be computed at every iteration of the broadcast. This means that in _some_ situations, _particularly_ in the presence of composed or multidimensional broadcasts, an expression with dot calls _may_ be computing a function more times than _intended_.

Is (at least to my eyes) already using a pretty neutral tone with plenty of hedging language. I think any negative framing here is already going to be pretty mitigated by its location on the page, being just below the antithetical tip `More dots: Fuse vectorized operations.` Of course there are many places where the described behavior is desired (like your `rand.()` example), but I don't think it's necessary to call these out explicitly.

If the comment was directed at the warning though, I'm open to any changes there. I think something, at least, should be communicated about the aliasing potential, but I wasn't sure what or where was the right place for a note like that.

Reply:
Sorry I meant on the bit in the docs introducing broadcasting!

Reply:
sounds good. I'll try another framing